### ISSUE TITLE
Error during sending staging table to Carol

### ISSUE THREAD

**Issue by RWallan** [totvslabs/pyCarol#607]
The following error has been reported by a client when trying to send a staging table to Carol.

```bash
Traceback (most recent call last):
  File "D:\GitHub\aulas-sc\bayer_export\main.py", line 35, in <module>
    carol_client.enviar_dados(
  File "D:\GitHub\aulas-sc\bayer_export\carol_client.py", line 69, in enviar_dados
    self.stag.send_data(
  File "D:\GitHub\aulas-sc\.venv\lib\site-packages\pycarol\staging.py", line 152, in send_data
    self.create_schema(_sample_json, staging_name, connector_id=connector_id, export_data=carol_data_storage,
  File "D:\GitHub\aulas-sc\.venv\lib\site-packages\pycarol\staging.py", line 293, in create_schema
    schema = schema.to_dict(mdmStagingType=staging_name, mdmFlexible=mdm_flexible,
  File "D:\GitHub\aulas-sc\.venv\lib\site-packages\pycarol\schema_generator.py", line 118, in to_dict
    schema_dict["mdmStagingMapping"]["properties"].update({prop: _dictConstructor(base_object=value)})
  File "D:\GitHub\aulas-sc\.venv\lib\site-packages\pycarol\schema_generator.py", line 137, in _dictConstructor
    schema_type = Type.get_schema_type_for(base_object_type)
  File "D:\GitHub\aulas-sc\.venv\lib\site-packages\pycarol\schema_generator.py", line 55, in get_schema_type_for
    np.float: DoubleType,
  File "D:\GitHub\aulas-sc\.venv\lib\site-packages\numpy\__init__.py", line 305, in __getattr__
    raise AttributeError(__former_attrs__[attr])
AttributeError: module 'numpy' has no attribute 'float'.
`np.float` was a deprecated alias for the builtin `float`. To avoid this error in existing code, use `float` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.float64` here.
The aliases was originally deprecated in NumPy 1.20; for more details and guidance see the original release note at:
    https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations. Did you mean: 'cfloat'?
```

I've already checked if the variables types is "str". And the error still ocurring.

The actuals versions of packages is:

```bash
[tool.poetry.dependencies]
python = "^3.10"
pycarol = "2.54.11"
unidecode = "^1.3.6"
pandas = "^1.5.3"
openpyxl = "^3.1.2"
```

And the `numpy` version is `1.24.3`.

In module schema_generator.py has the `SCHEMA_TYPES` dict inside `Type` class with this structure:

```python
class Type(object):
    @classmethod
    def get_schema_type_for(cls, t):
        import pandas as pd
        import numpy as np

        """docstring for get_schema_type_for"""

        SCHEMA_TYPES = {
            type(None): NullType,
            str: StringType,
            int: IntType,
            float: DoubleType,
            bool: BooleanType,
            list: ArrayType,
            dict: ObjectType,
            np.float64: DoubleType,
            np.float32: DoubleType,
            np.float: DoubleType,
            np.int64: IntType,
            np.int32: IntType,
            np.int: IntType,
            np.bool_: BooleanType,
            datetime: DateType,
            pd.Timestamp:DateType
        }

        schema_type = SCHEMA_TYPES.get(t)

        if not schema_type:
            raise JsonSchemaTypeNotFound("There is no schema type for  %s.\n Try:\n %s" % (
            str(t), ",\n".join(["\t%s" % str(k) for k in SCHEMA_TYPES.keys()])))
        return schema_type
```

I don't think it's necessary to keep np.float and np.int because in the updated versions of numpy there is no such type.

**Comment by rafarui** [comment]
we need to keep it, otherwise it will break compatibility with old np versions that might be using these types. 

we could fallback to string if it is not defined, or check for those two types before creating the dict.

**Comment by RWallan** [comment]
A suggestion is maybe to add a try/except to check if these types exist or not. Like this, perhaps?

```python
class Type(object):
    @classmethod
    def get_schema_type_for(cls, t):
        import pandas as pd
        import numpy as np

        """docstring for get_schema_type_for"""

        try:
            np.float
            np.int
        except AttributeError:
            np.float = np.float64
            np.int = np.int64

        SCHEMA_TYPES = {
            type(None): NullType,
            str: StringType,
            int: IntType,
            float: DoubleType,
            bool: BooleanType,
            list: ArrayType,
            dict: ObjectType,
            np.float64: DoubleType,
            np.float32: DoubleType,
            np.float: DoubleType,
            np.int64: IntType,
            np.int32: IntType,
            np.int: IntType,
            np.bool_: BooleanType,
            datetime: DateType,
            pd.Timestamp:DateType
        }

        schema_type = SCHEMA_TYPES.get(t)

        if not schema_type:
            raise JsonSchemaTypeNotFound("There is no schema type for  %s.\n Try:\n %s" % (
            str(t), ",\n".join(["\t%s" % str(k) for k in SCHEMA_TYPES.keys()])))
        return schema_type
```

**Comment by darolt** [comment]
fixed on pycarol 2.55.0.

**Comment by RWallan** [comment]
Thanks!
